src/main/java/org/broadinstitute/hellbender/engine/ReadShard.java

```diff
@@ -36,12 +37,11 @@
      *
      * @param interval the genomic span covered by this shard
      * @param paddedInterval the span covered by this shard, plus any additional padding on each side (must contain the un-padded interval)
-     * @param readsSource source of reads from which to populate this shard
+     * @param readsSource source of reads from which to populate this shard; it could be null if no data source is available
```

Comment:
I think it's ok if you want to allow `ReadShards` to have a null data source, provided you add a `public boolean hasBackingDataSource()` method that returns false when `readsSource == null` to allow the client to query this.

It seems strange, though, to imagine a `SlidingWindowWalker` that does not take reads, and yet takes `ReadShards` in its `apply()` method. I wonder if perhaps we need a `Shard` superclass for `ReadShard` that contains only `interval` and `paddedInterval`? Then your apply method could take a `Shard, ReadsContext, ReferenceContext, FeatureContext`. Or are you happy with the existing implementation?